_docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme

```diff
@@ -413,79 +355,56 @@ function rhdp2_preprocess_assembly(&$variables) {
         $audience_selection_values[] = $field_audience_selection_value['value'];
       }
 
+      // @TOD: This should be passed to the attributes array.
       $audience_selection = implode(',', $audience_selection_values);
       $variables['audience_selection'] = " data-audience={$audience_selection}";
     }
   }
 
-  $variables['background_image_url'] = FALSE;
-  $image_value = FALSE;
-  $svg_value = FALSE;
-
-  if ($assembly->hasField('field_background_image')) {
-    $image_value = $assembly->get('field_background_image')->getValue();
-  }
-
-  if ($assembly->hasField('field_background_svg')) {
-    $svg_value = $assembly->get('field_background_svg')->getValue();
+  // Set the background image url from the field.
+  if ($assembly->hasField('field_background_image') && !$assembly->get('field_background_image')->isEmpty()) {
+    $file_id = $assembly->field_background_image->target_id;
+    $uri = File::load($file_id)->getFileUri();
+    $variables['background_image_url'] = ImageStyle::load('billboard')->buildUrl($uri);
+    $variables['attributes']->addClass('has-background');
   }
 
-  if ($svg_value) {
-    $file_id = $svg_value[0]['target_id'];
+  // Set the background image url for svg.
+  // @TOOD: If an assembly has both, does the SVG win?
+  if ($assembly->hasField('field_background_svg') && !$assembly->get('field_background_svg')->isEmpty()) {
+    $file_id = $assembly->field_background_svg->target_id;
     $uri = File::load($file_id)->getFileUri();
     $variables['background_image_url'] = file_create_url($uri);
     $variables['attributes']->addClass('has-background');
-  } else if ($image_value) {
-    $file_id = $image_value[0]['target_id'];
-    $uri = File::load($file_id)->getFileUri();
-    $variables['background_image_url'] = ImageStyle::load('billboard')->buildUrl($uri);
-    $variables['attributes']->addClass('has-background');
   }
 
+  // Set the image_uri and image_alt variables.
   if ($assembly->hasField('field_image')) {
     if ($image_value = $assembly->get('field_image')->getValue()) {
       if ($file_id = $image_value[0]['target_id']) {
```

Comment:
I think that we need to verify that `$image_value[0]['target_id]` is set before accessing it

Reply:
Nope, that is not true. `$assembly->get('field_image')->getValue()` will evaluate to `array{}` which is false-y, so this evaluates as we expected

Reply:
I didn't spend as much time in preprocess_assembly as much as I did preprocess_node.   If you would like me to work on that hook after your other comments, I will do that.